javascript/widgets/handler/passwordsignup.js

```diff
@@ -69,15 +70,22 @@ firebaseui.auth.widget.handler.handlePasswordSignUp = function(
  * @private
  */
 firebaseui.auth.widget.handler.onSignUpSubmit_ = function(app, component) {
+  var requireDisplayName = app.getConfig().getRequireDisplayName();
+
   // Check fields are valid.
   var email = component.checkAndGetEmail();
-  var name = component.checkAndGetName();
+
+  var name = null;
```

Comment:
If you look [here](https://github.com/firebase/firebaseui-web/blob/3cdef074eb095b6d5ce77bf59c1b9f0f0eb4c2e5/javascript/widgets/handler/passwordsignup.js#L109) you'll see that we do an extra RPC request to set the `displayName`. If we are not requiring `displayName`, then this call should not be done.

You should change the logic to not `updateProfile` when `requireDisplayName` is `false`.